pkg/kudoctl/cmd/upgrade.go

```diff
@@ -0,0 +1,147 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/Masterminds/semver"
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1alpha1"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/cmd/install"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/kudo"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/repo"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var (
+	upgradeExample = `
+		The upgrade argument must be a name of the package in the repository, a path to package in *.tar.gz format,
+		or a path to an unpacked package directory.
+
+		# Upgrade flink instance dev-flink to the latest version
+		kubectl kudo upgrade flink --instance dev-flink
+		
+		*Note*: should you have a local "flink" folder in the current directory it will take precedence over the remote repository.
+
+		# Upgrade flink to the version 1.1.1
+		kubectl kudo upgrade flink --instance dev-flink --version 1.1.1`
+)
+
+type options struct {
+	InstanceName   string
+	Namespace      string
+	PackageVersion string
+}
+
+// defaultOptions initializes the install command options to its defaults
+var defaultOptions = &options{
+	Namespace: "default",
+}
+
+// newUpgradeCmd creates the install command for the CLI
+func newUpgradeCmd() *cobra.Command {
+	options := defaultOptions
+	upgradeCmd := &cobra.Command{
+		Use:     "upgrade <name>",
+		Short:   "-> Upgrade KUDO package.",
+		Long:    `Upgrade KUDO package from current version to new version.`,
+		Example: upgradeExample,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runUpgrade(args, options)
+		},
+		SilenceUsage: true,
+	}
+
+	upgradeCmd.Flags().StringVar(&options.InstanceName, "instance", "", "The instance name. (default to Operator name)")
```

Comment:
(default to Operator name) may make sense for `install`.   It doesn't make sense here for upgrade.